pytorch_lightning/utilities/apply_func.py

```diff
@@ -161,14 +165,14 @@ def batch_to(data):
     return apply_to_collection(batch, dtype=dtype, function=batch_to)
 
 
-def convert_to_tensors(data, device: torch.device = None):
+def convert_to_tensors(data: Any, device: Union[str, torch.device] = None) -> Any:
     if device is None:
         raise MisconfigurationException("device (torch.device) should be provided.")
 
```

Comment:
why do we set the device to None as the default and then error if it's not set? 
Should we remove the default value? or type it as Optional?

Reply:
Yes, we could.